pkg/capabilities/capabilities.go

```diff
@@ -68,11 +68,13 @@ func Setup(allowPrivileged bool, perConnectionBytesPerSec int64) {
 	})
 }
 
-// SetForTests sets capabilities for tests.  Convenience method for testing.  This should only be called from tests.
-func SetForTests(c Capabilities) {
+// ResetForTest resets the capabilities to a given state for testing purposes.
+// This function should only be called from tests.
+func ResetForTest(c *Capabilities) {
 	capInstance.lock.Lock()
 	defer capInstance.lock.Unlock()
-	capInstance.capabilities = &c
+	capInstance.capabilities = c
+	capInstance.once = sync.Once{}
```

Comment:
this function is confusing now, and combines reset AND Initialize/Setup

if we want a reset function, and for all sets to go through the normal code path, don't let this function take a `c *Capabilities` parameter, just make this function reset:

```go
func ResetForTest(c *Capabilities) {
	capInstance.lock.Lock()
	defer capInstance.lock.Unlock()
	capInstance.capabilities = nil
	capInstance.once = sync.Once{}
```

Reply:
Hi @liggitt 
I've refactored the method as you suggested, so now it is truly only resetting the capInstance and not doing anything else. For intialize/setup, I am now using the Initialize method so as to keep to the normal code path as much as possible. Thanks for the suggestion!